From: IrfanMohammad <irfan@graphiant.com>
Date: Sun, 27 Aug 2023 03:42:04 +0000
Subject: increase timeout for cb and upgr mod locks

Upgrade lock is held for longer as callbacks need to finish processing
before we can downgrade to a read lock.
---
 src/common.h  | 4 ++--
 src/shm_mod.c | 7 ++++++-
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/src/common.h b/src/common.h
index e05e9da..663859e 100644
--- a/src/common.h
+++ b/src/common.h
@@ -126,10 +126,10 @@ struct srplg_ntf_s;
 #define SR_SHMEXT_SUB_LOCK_TIMEOUT 15000
 
 /** default timeout for change subscription callback (ms) */
-#define SR_CHANGE_CB_TIMEOUT 5000
+#define SR_CHANGE_CB_TIMEOUT 15000
 
 /** default timeout for operational subscription callback (ms) */
-#define SR_OPER_CB_TIMEOUT 5000
+#define SR_OPER_CB_TIMEOUT 10000
 
 /** default timeout for RPC/action subscription callback (ms) */
 #define SR_RPC_CB_TIMEOUT 2000
diff --git a/src/shm_mod.c b/src/shm_mod.c
index b0ff875..5c3961f 100644
--- a/src/shm_mod.c
+++ b/src/shm_mod.c
@@ -1317,7 +1317,12 @@ sr_shmmod_lock(struct sr_mod_lock_s *shm_lock, uint32_t timeout_ms, sr_lock_mode
 
     if (!timeout_ms) {
         /* default timeout */
-        timeout_ms = SR_MOD_LOCK_TIMEOUT;
+        if (SR_LOCK_READ_UPGR == mode) {
+            /* upgr needs to wait longer than it takes for an apply_changes to complete */
+            timeout_ms = 3 * SR_MOD_LOCK_TIMEOUT;
+        } else {
+            timeout_ms = SR_MOD_LOCK_TIMEOUT;
+        }
     }
 
 ds_lock_retry:
